File: client/src/application/semesters/dto/semester.dto.js
class SemesterDTO {
  constructor(data = {}) {
    this.semesterId = data.semester_id;
    this.name = data.name;
    this.semesterNumber = data.semester_number;

    this.academicYearId = data.academic_year_id;
    this.academicYearName = data.academic_year_name;

    this.universityId = data.university_id;
    this.universityName = data.university_name;

    this.startDate = data.start_date;
    this.endDate = data.end_date;

    this.status = data.status;
  }
}

module.exports = SemesterDTO;

File: server/apps/semesters/application/dto/semester_dto.py
from dataclasses import dataclass
from datetime import date


@dataclass
class SemesterDTO:

    semester_id: str
    name: str
    semester_number: str
    academic_year_id: str
    academic_year_name: str
    university_id: str
    university_name: str
    start_date: date
    end_date: date
    status: str